network/unix_domain_server_test.cc

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright 2025 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "unix_domain_server.h"
+
+#include <gtest/gtest.h>
+
+namespace Network
+{
+namespace
+{
+
+constexpr char kTestServerAddress[] = "test_server_socket";
+
+// The tests assume the client and server are running on a PC (Linux), as we are not using
+// an Android device. However, this creates a limitation: we cannot directly test the TcpClient
+// against the UnixDomainServer because they use incompatible communication protocols. The TcpClient
+// uses TCP/IP sockets, while the UnixDomainServer uses Unix domain sockets (UDS). To test the
+// UnixDomainServer, we will start an instance of it and use a basic, raw socket client to verify
+// its behavior, such as handling connections and basic messages.
+
+// Creates a raw socket client and connects to the Unix Domain Server.
+absl::StatusOr<std::unique_ptr<SocketConnection>> ConnectClient()
+{
+    SocketType client_socket = socket(AF_UNIX, SOCK_STREAM, 0);
+    if (client_socket < 0)
+    {
+        return absl::InternalError("Client: socket creation failed.");
+    }
+    /// Connects to an Unix (Local) Domain with abstract namespace.
+    sockaddr_un addr = {};
+    addr.sun_family = AF_UNIX;
+    // first char is '\0'
+    addr.sun_path[0] = '\0';
+    strncpy(addr.sun_path + 1, kTestServerAddress, std::size(kTestServerAddress));
+    if (connect(client_socket,
+                (sockaddr*)&addr,
+                offsetof(sockaddr_un, sun_path) + std::size(kTestServerAddress)) < 0)
+    {
+        close(client_socket);
+        return absl::UnavailableError("Client: connect failed.");
+    }
+
+    absl::StatusOr<std::unique_ptr<SocketConnection>> connection = SocketConnection::Create(
+    client_socket);
+    if (!connection.ok())
+    {
+        return connection.status();
+    }
+    return *std::move(connection);
+}
+
+TEST(UnixDomainServerTest, StartAndStop)
+{
+    auto server = std::make_unique<UnixDomainServer>();
+    ASSERT_TRUE(server->Start(kTestServerAddress).ok());
+    server->Stop();
+    // Should return immediately.
+    server->Wait();
+}
+
+TEST(UnixDomainServerTest, ClientConnectAndDisconnect)
+{
+    auto server = std::make_unique<UnixDomainServer>();
+    ASSERT_TRUE(server->Start(kTestServerAddress).ok());
+
+    absl::StatusOr<std::unique_ptr<SocketConnection>> client_conn = ConnectClient();
+    ASSERT_TRUE(client_conn.ok());
```

Comment:
Prefer using the `absl_testing::IsOk()` matcher since it will print more information if something fails
```suggestion
#include "absl/status/status_matchers.h"
using ::absl_testing::IsOk;
// ...
    ASSERT_THAT(client_conn.ok(), IsOk());
```